src/index.ts

```diff
@@ -20,3 +20,4 @@ export { makeFetchClient, SoFetchResponse }
 export default defaultFetch
 
 export * from './interfaces'
+export { default as SoFetch } from './so-fetch';
```

Comment:
It seems like you did not really want to expose this class for some reason, but to be able to use it properly with typescript we need to expose it, or create a specific interface/type of it to actually expose which feels like more work anyway.